src/codegen/mod.rs

```diff
@@ -84,6 +84,11 @@ impl CodeGenerator {
                                 module.push_function(func);
                             }
                             ClassMember::Method(method) => {
+                                let mut method = method.clone();
+                                method.parameters.insert(
+                                    0,
+                                    Parameter::new("self", ParsedType::TypeName(c.name.clone())),
+                                );
```

Comment:
@YanHerChen Here transform object-orient to pure procedure-orient code. Since IR don't support concept **class**